Core/Core/Serialisation/BaseObjectDeserializerV2.cs

```diff
@@ -161,7 +176,15 @@ private List<(string, int)> GetClosures(string rootObjectJson)
       doc1 = JObject.Load(reader);
     }
 
-    object? converted = ConvertJsonElement(doc1);
+    object? converted;
+    try
+    {
+      converted = ConvertJsonElement(doc1);
+    }
+    catch (Exception ex)
+    {
+      throw new SpeckleException($"Failed to deserialize {doc1} as {doc1.Type}", ex);
+    }
```

Comment:
I see what you're doing here... I'm assuming this was out of scope for the current ticket and it was cleaner to wrap it in a SpeckleException and move the fix to the `Exception Warning` effort. Is this correct?

If so, we can live with it.

Reply:
OH this makes so much sense... 🤐 I'll accept it, and maybe we should keep this as a good example for the team too!

@BovineOx would love your take on this too just to be sure we're all on the same page.

Reply:
My only concern would be if there are critical failure that we would be masking by wrapping this, probably OOM springs to mind. We could create some WrapIfNotTerminal helper that would re-throw if the type of the exception was not one we wanted to mask, but I otherwise think it's fine.

Reply:
Agreed with @BovineOx, the OOM is a great illustration.

It would be preferable to catch specific exception here, but our serializer is quite the beast, with no design towards exceptions (in that we call and pray) so I'm hesitant to improve on this much further (I'd be happier to re-write our serializer with more explicit ownership of exceptional paths)

---
The `WrapIfNotTerminal` is an interesting idea. Although, if I may iterate slightly; something like this?

```cs
if(IsTerminal(ex))
{
   throw;
}
throw new SpeckleException("...", ex);
```

However, this shouldn't be a substitute to good code design, so I'm hesitant to jump on this too.

Reply:
So, what do we think, any action needed for this PR?

Reply:
We probably pollute the stack less with the inline check but I am bit meh about having that everywhere..

It might be better to have some TryThrowWrapper, but how do we feel about that, it would be perhaps... TryThrowWrapper(<operation>, Action), something like...

Utils.TryThrowWrapper("Create Toposolid", () =>
{
  // my action
});

And internally would handle the catches nicely and throw he new SpeckleException($"Action: {action}", ex);

I think it's worth picking something, implementing it here and then advertising it as a use this going forwards for these types of cases and then we have one place to augment, tweak it.

Thoughts?

Reply:
Resolving this as the change is incoming https://github.com/specklesystems/speckle-sharp/pull/3114